src/bin/vip-slowlogs.js

```diff
@@ -0,0 +1,218 @@
+	} catch ( error ) {
+		await trackEvent( 'slowlogs_command_error', { ...trackingParams, error: error.message } );
+
+		return exit.withError( error.message );
+	}
+
+	await trackEvent( 'slowlogs_command_success', {
+		...trackingParams,
+		total: slowlogs.nodes.length,
+	} );
+
+	if ( ! slowlogs.nodes.length ) {
+		console.error( 'No logs found' );
+		return;
+	}
+
+	printSlowlogs( slowlogs.nodes, opt.format );
+}
+
+export async function followLogs( opt ): Promise< void > {
+	let after = null;
+	let isFirstRequest = true;
+	// How many times have we polled?
+	let requestNumber = 0;
+
+	const trackingParams = getBaseTrackingParams( opt );
+
+	// Set an initial default delay
+	let delay = DEFAULT_POLLING_DELAY_IN_SECONDS;
+
+	// eslint-disable-next-line no-constant-condition
+	while ( true ) {
+		const limit = isFirstRequest ? opt.limit : LIMIT_MAX;
+
+		requestNumber++;
+		trackingParams.request_number = requestNumber;
+		trackingParams.request_delay = delay;
+		trackingParams.limit = limit;
+
+		let slowlogs;
+		try {
+			// eslint-disable-next-line no-await-in-loop
+			slowlogs = await slowlogsLib.getRecentSlowlogs( opt.app.id, opt.env.id, limit, after );
+
+			// eslint-disable-next-line no-await-in-loop
+			await trackEvent( 'slowlogs_command_follow_success', {
+				...trackingParams,
+				total: slowlogs?.nodes.length,
+			} );
+		} catch ( error ) {
+			// eslint-disable-next-line no-await-in-loop
+			await trackEvent( 'slowlogs_command_follow_error', {
+				...trackingParams,
+				error: error.message,
+			} );
+
+			// If the first request fails we don't want to retry (it's probably not recoverable)
+			if ( isFirstRequest ) {
+				console.error( `${ chalk.red( 'Error:' ) } Failed to fetch slowlogs.` );
+				break;
+			}
+			// Increase the delay on errors to avoid overloading the server, up to a max of 5 minutes
+			delay += DEFAULT_POLLING_DELAY_IN_SECONDS;
+			delay = Math.min( delay, MAX_POLLING_DELAY_IN_SECONDS );
+			console.error(
+				`${ chalk.red( 'Error:' ) } Failed to fetch slowlogs. Trying again in ${ delay } seconds.`
+			);
+		}
+
+		if ( slowlogs ) {
+			if ( slowlogs?.nodes.length ) {
+				printSlowlogs( slowlogs.nodes, opt.format );
+			}
+
+			after = slowlogs?.nextCursor;
+			isFirstRequest = false;
+
+			// Keep a sane lower limit of MIN_POLLING_DELAY_IN_SECONDS just in case something goes wrong in the server-side
+			delay = Math.max(
+				slowlogs?.pollingDelaySeconds || DEFAULT_POLLING_DELAY_IN_SECONDS,
+				MIN_POLLING_DELAY_IN_SECONDS
+			);
+		}
+
+		// eslint-disable-next-line no-await-in-loop
+		await setTimeout( delay * 1000 );
+	}
+}
+
+function getBaseTrackingParams( opt ) {
+	return {
+		command: 'vip slowlogs',
+		org_id: opt.app.organization.id,
+		app_id: opt.app.id,
+		env_id: opt.env.id,
+		limit: opt.limit,
+		follow: opt.follow || false,
+		format: opt.format,
+	};
+}
+
+function printSlowlogs( slowlogs, format ) {
```

Comment:
There is a build-in formatting feature to vip-cli already. Could we use that instead to stay consistent? Or is your format too different?

Example of usage:
https://github.com/Automattic/vip-cli/blob/update/sync_error_handling/src/bin/vip-config-envvar-get-all.js#L77

Reply:
@pschoffer Good catch, I'll test it with the build-in format. It should works well :)

Reply:
@pschoffer done, thanks :)